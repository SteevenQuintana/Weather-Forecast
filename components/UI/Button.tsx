import { ReactNode } from 'react'

interface ButtonProps {
  children: ReactNode
  type?: ButtonType
  className?: string
  onClick?: () => void
}

type ButtonType = 'button' | 'submit' | 'reset'

const Button = ({
  children,
  type = 'button',
  className,
  onClick
}: ButtonProps) => {
  return (
    <button
      type={type}
      className={`bg-[#ffffff] dark:bg-[#1c2f3d] text-[#1d1d1d] dark:text-[#dbdbdb] rounded-md py-2 px-7 ${className}`}
      onClick={onClick}
    >
      {children}
    </button>
  )
}

export default Button
