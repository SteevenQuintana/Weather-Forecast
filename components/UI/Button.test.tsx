import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import Button from './Button'

describe('Button', () => {
  it('should render a button with correct children', () => {
    render(<Button>Click me</Button>)

    const button = screen.getByRole('button', {
      name: /Click me/i
    })

    expect(button).toBeInTheDocument()
  })

  it('should render a button with correct type attribute', () => {
    render(<Button type='submit'>Submit</Button>)
    const button = screen.getByText('Submit')
    expect(button).toHaveAttribute('type', 'submit')
  })

  it('should call onClick function when the button is clicked', () => {
    const onClick = jest.fn()
    render(<Button onClick={onClick}>Click me</Button>)
    const button = screen.getByRole('button', {
      name: /Click me/i
    })
    button.click()
    expect(onClick).toHaveBeenCalled()
  })

  it('should match the snapshot', () => {
    const { container } = render(<Button>Click me</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })
})
