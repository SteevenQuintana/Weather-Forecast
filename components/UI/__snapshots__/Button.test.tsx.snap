// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
<button
  class="bg-[#ffffff] dark:bg-[#1c2f3d] text-[#1d1d1d] dark:text-[#dbdbdb] rounded-md py-2 px-7 undefined"
  type="button"
>
  Click me
</button>
`;
